File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        fst = snd = float('inf')
        for num in nums:
            if num<=fst:
                fst = num
            elif num <= snd:
                snd = num
            else:
                return True
        return False
        
        
        
        

File: 334-increasing-triplet-subsequence/increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int fst = Integer.MAX_VALUE;
        int snd = Integer.MAX_VALUE;
        for(int n : nums) {
            if(n <= fst) {
                fst = n;
            } else if(n <= snd) {
                snd = n;
            } else {
                return true;
            }
        }
        return false;
    }
}